tests/unit/test_psycopg2_connector.py

```diff
@@ -14,15 +13,54 @@ def func():
         func()
 
 
-@pytest.mark.asyncio
-async def test_wrap_exceptions_success():
+def test_wrap_exceptions_success():
     @psycopg2_connector.wrap_exceptions
     def func(a, b):
         return a, b
 
     assert func(1, 2) == (1, 2)
 
 
+def test_wrap_query_exceptions_reached_max_tries(mocker):
+    @psycopg2_connector.wrap_query_exceptions
+    def func(connector):
+        raise psycopg2.errors.InterfaceError("connection already closed")
+
+    connector = mocker.Mock(_pool=mocker.Mock(maxconn=5))
+
+    with pytest.raises(exceptions.ConnectorException) as excinfo:
+        func(connector)
+
+    assert str(excinfo.value) == "Could not get a valid connection after 6 tries"
+
+
+@pytest.mark.parametrize("exception_class", [Exception, psycopg2.errors.InterfaceError])
+def test_wrap_query_exceptions_unhandled_exception(mocker, exception_class):
+    @psycopg2_connector.wrap_query_exceptions
+    def func(connector):
+        raise exception_class("foo")
+
+    connector = mocker.Mock(_pool=mocker.Mock(maxconn=5))
+
+    with pytest.raises(exception_class):
+        func(connector)
+
+
+def test_wrap_query_exceptions_success(mocker):
+    cnt = 0
+
+    @psycopg2_connector.wrap_query_exceptions
+    def func(connector, a, b):
+        nonlocal cnt
+        if cnt < 2:
+            cnt += 1
```

Comment:
A more natural way to do this is to declare a list outside the function, and append things inside.

Reply:
Done.